firestore/integration_test_internal/src/settings_test.cc

```diff
@@ -94,6 +96,103 @@ TEST(SettingsTest, Equality) {
   EXPECT_TRUE(settings4 != settings5);
 }
 
+TEST(SettingsTest, EqualityWithLocalCacheSettings) {
+  constexpr int64_t kFiveMb = 5 * 1024 * 1024;
+  constexpr int64_t kSixMb = 6 * 1024 * 1024;
+
+  Settings settings1;
+  settings1.set_host("foo");
+  settings1.set_ssl_enabled(true);
+  settings1.set_local_cache_settings(
+      PersistentCacheSettings::Create().WithSizeBytes(kFiveMb));
+
+  Settings settings2;
+  settings2.set_host("bar");
+  settings2.set_ssl_enabled(true);
+  settings2.set_local_cache_settings(
+      PersistentCacheSettings::Create().WithSizeBytes(kFiveMb));
+
+  Settings settings3;
+  settings3.set_host("foo");
+  settings3.set_ssl_enabled(false);
+  settings3.set_local_cache_settings(
+      PersistentCacheSettings::Create().WithSizeBytes(kFiveMb));
+
+  Settings settings4;
+  settings4.set_host("foo");
+  settings4.set_ssl_enabled(true);
+  settings4.set_local_cache_settings(MemoryCacheSettings::Create());
+
+  Settings settings5;
+  settings5.set_host("foo");
+  settings5.set_ssl_enabled(true);
+  settings5.set_local_cache_settings(
+      PersistentCacheSettings::Create().WithSizeBytes(kSixMb));
+
+  Settings settings6;
+  settings6.set_host("foo");
+  settings6.set_ssl_enabled(true);
+  settings6.set_local_cache_settings(
+      MemoryCacheSettings::Create().WithGarbageCollectorSettings(
+          MemoryEagerGCSettings::Create()));
+
+  Settings settings7;
+  settings7.set_host("foo");
+  settings7.set_ssl_enabled(true);
+  settings7.set_local_cache_settings(
+      MemoryCacheSettings::Create().WithGarbageCollectorSettings(
+          MemoryLruGCSettings::Create().WithSizeBytes(kFiveMb)));
+
+  Settings settings8;
+  settings8.set_host("foo");
+  settings8.set_ssl_enabled(true);
+  settings8.set_local_cache_settings(
+      MemoryCacheSettings::Create().WithGarbageCollectorSettings(
+          MemoryLruGCSettings::Create().WithSizeBytes(kSixMb)));
+
+  // Same as settings7
+  Settings settings9;
+  settings9.set_host("foo");
+  settings9.set_ssl_enabled(true);
+  settings9.set_local_cache_settings(
+      MemoryCacheSettings::Create().WithGarbageCollectorSettings(
+          MemoryLruGCSettings::Create().WithSizeBytes(kFiveMb)));
+
+  EXPECT_TRUE(settings1 == settings1);
+  EXPECT_TRUE(settings6 == settings4);
```

Comment:
Shouldn't `settings6` and `settings4` compare _unequal_ since they have different persistence settings?